src/Chapter4.hs

```diff
@@ -629,7 +652,7 @@ Can you implement a monad version of AND, polymorphic over any monad?
 🕯 HINT: Use "(>>=)", "pure" and anonymous function
 -}
 andM :: (Monad m) => m Bool -> m Bool -> m Bool
-andM = error "andM: Not implemented!"
+andM mx my = mx >>= (\y -> (y&&) <$> my) 
```

Comment:
This definition doesn't really use the full power of Monads. It's equivalent to:

```haskell
fmap (&&) mx <*> my
```

And if you don't use the `>>=` operator, then the `Monad` constraint is too strong, and it is enough to specify only `Applicative`. However, with `Applicative` you are not able to implement a lazier version of `andM`.

The ordinary `&&` if lazy on the second argument. Meaning that if it meets `False` as the first argument, there is no need to calculate another arg. In `andM` we would like to keep the same behaviour, so the

```
Just False `andM` _
```

should be `Just False` :slightly_smiling_face:

Reply:
>  Meaning that if it meets False as the first argument, there is no need to calculate another arg...
This cleared the confusion:

I was working on this example:

If I understand your explanation correctly, then the output of the expression below should be correct?
```
>>> [False,True,False] `andM` [True,False,True]

Correct Output :  [False,True,False,True,False]
```
I was trying to get this incorrect output `[False,False,False,True,False,True,False,False,False]` where the `&&` operation is performed between each element inside both lists.

Reply:
Yep, that's the correct output!